core/http/src/main/java/org/trellisldp/http/WebAcFilter.java

```diff
@@ -149,6 +151,14 @@ public void filter(final ContainerRequestContext req, final ContainerResponseCon
         }
     }
 
+    private boolean reqAudit(final ContainerRequestContext ctx) {
+        final Prefer prefer = Prefer.valueOf(ctx.getHeaderString(PREFER));
```

Comment:
Just a thought, maybe change to `Optional<Prefer> Prefer::valueOf`? Or add a method under a new name, since `static valueOf` is pretty well-known from `enum`s.

Reply:
`Prefer::valueOf` will need to stay as it is in order to play nicely with the JAX-RS header parsing mechanism. Plus, aren't we trying to avoid all those `Optional` types? :-)

Reply:
Not at all! This is [the exact use case](https://stackoverflow.com/a/26328555) for which they were meant. In any event, it was just a thought-- no need to hold this PR up.

Reply:
I agree with that use case. What I'm referring to, though, is the use of [`javax.ws.rs.HeaderParam`](https://docs.oracle.com/javaee/7/api/javax/ws/rs/HeaderParam.html). Note the third item in the list of requirements:

> Have a static method named `valueOf` or `fromString` that accepts a single `String` argument (see, for example, `Integer.valueOf(String)`)

Reply:
Sure, that's why I would "add a method under a new name" a la `Optional<Prefer> parse(String)` or the like. But it's that much more code, so unless you expect to do this a lot, probably easier to just do what you do here.

Reply:
Right, I don't expect to be doing this sort of thing very much.